lib/__tests__/Button.test.js

```diff
@@ -52,6 +52,21 @@ describe('<Button>', function() {
     expect(shallowToJson(wrapper)).toMatchSnapshot();
   });
 
+  it('DOM with some props and component instance used instead of tag', function() {
+    var button = el(
+      Button,
+      {
+        id: 'foo',
+        className: 'bar',
+        tag: 'div',
+        component: el('button'),
```

Comment:
We need this test to verify behavior with a non-standard component. `button` can already be covered by the existing functionality: let's test the extent of the new functionality.